src/seedu/addressbook/logic/Logic.java

```diff
@@ -76,7 +76,7 @@ public CommandResult execute(String userCommandText) throws Exception {
     }
 
     /**
-     * Executes the command, updates storage, and returns the result.
+     * Executes the command, updates storage if data will be mutated, and returns the result.
```

Comment:
Good job on updating the header comments to reflect the change in behaviour of `Logic`!